### PR TITLE
feat: refactor extensions as npm packages (be brave 1)

### PR DESCRIPTION
**Pull request by marrouchi**
# Motivation

This PR, we allow for extensions to be installed as npm packages :
- Added parent class Extension, so that Channel, Helper and Plugin would extend from it
- Each Extension will load it's own i18n translations
- Align methods : Name, Namespaces
- Add extension suffix : -plugin, -channel, -helper
- Enforce typing

# Type of change:

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/src/chat/controllers/block.controller.ts [hunk 1]
```diff
@@ -122,11 +125,12 @@ export class BlockController extends BaseController<
       const plugins = this.pluginsService
         .getAllByType(PluginType.block)
         .map((p) => ({
-          id: p.id,
+          id: p.getName(),
```
[the remark this answers is not in the document]
Comment (marrouchi) [reply api/src/chat/controllers/block.controller.ts:128]: I believe that has been updated.

File: api/src/chat/controllers/block.controller.ts [hunk 2]
```diff
@@ -85,20 +85,23 @@ export class BlockController extends BaseController<
...
    * @returns An array containing the settings of the specified plugin.
    */
   @Get('customBlocks/settings')
-  findSettings(@Query('plugin') pluginId: string) {
+  findSettings(@Query('plugin') pluginName: PluginName) {
     try {
-      if (!pluginId) {
+      if (!pluginName) {
         throw new BadRequestException(
           'Plugin id must be supplied as a query param',
```
[the remark this answers is not in the document]
Comment (marrouchi) [reply api/src/chat/controllers/block.controller.ts:97]: Thanks

**Review by yassinedorbozgithub — APPROVED** [review]
I have tested :
:white_check_mark: Blocks
:white_check_mark: Ollama default LLM helper
:white_check_mark: Widget integration + wordpress

LGTM :+1:

**Review by IkbelTalebHssan — APPROVED** [review]

**Review by MohamedAliBouhaouala — APPROVED** [review]
Looks good to me.
Just quick remark: we need to update the documentation accordingly. Perhaps add a future ticket for resolving the "npx hexabot install" command issue? Some potential enhancement may also include uninstalling plugins too.

**Pull request merged**